autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java

```diff
@@ -9,45 +9,64 @@
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.InputStream;
 import java.net.MalformedURLException;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 @Getter
 public class MavenVersionRepository implements VersionRepository {
+    private static final transient int FILENOTFOUNDSTATUSCODE = 404;
     private final String baseUrl;
+    private final HttpClient httpClient;
 
     public MavenVersionRepository(String baseUrl) {
         this.baseUrl = baseUrl;
+        this.httpClient = HttpClient.newBuilder()
+                .connectTimeout(Duration.ofSeconds(2))
+                .followRedirects(HttpClient.Redirect.NEVER)
+                .version(HttpClient.Version.HTTP_2)
+                .build();
     }
 
     @Override
     public Set<Version> getAllAvailableVersions(Dependency dependency) {
-        try(BufferedReader in =
-                    new BufferedReader(
-                            new InputStreamReader(
-                                    getRepoUri(dependency)
-                                            .openConnection()
-                                            .getInputStream()
-                            )
-                    )
-        ) {
-            return new MetadataXpp3Reader()
-                    .read(in)
-                    .getVersioning()
-                    .getVersions()
-                    .stream()
-                    .map(MavenVersion::new)
-                    .collect(Collectors.toUnmodifiableSet());
-        } catch (XmlPullParserException e) {
-            throw new DependencyParserException("something went wrong while parseing the xml", e);
-        }catch (IOException e) {
-            return new HashSet<>();
+        try {
+            HttpResponse<InputStream> response = getInputStreamHttpResponse(dependency);
+            if (response.statusCode() == FILENOTFOUNDSTATUSCODE){
+                return new HashSet<>();
+            }
+            else {
+                return new MetadataXpp3Reader()
+                        .read(response.body())
+                        .getVersioning()
+                        .getVersions()
+                        .stream()
+                        .map(MavenVersion::new)
+                        .collect(Collectors.toUnmodifiableSet());
+            }
+        } catch (XmlPullParserException | IOException e) {
+            throw new DependencyParserException("something went wrong while parsing the xml", e);
+        }
+    }
+
+    private HttpResponse<InputStream> getInputStreamHttpResponse(Dependency dependency) {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().GET().uri(getRepoUri(dependency).toURI())
+                    .build();
```

Comment:
Whenever sending an http request it is a good practice to define a timeout how long you want to wait for a response. E.g. wait at most 5 seconds for a response:

`.uri(...).timeout(Duration.ofSeconds(5)).build()`